### ISSUE TITLE
Perform Batch Normalization

### ISSUE THREAD

**Issue by manuelli** [RobotLocomotion/pytorch-dense-correspondence#57]
Per discussion with Yunzhu think about how to get batch normalization to work with out setup.

One potential idea.
- Sample N pairs of images (this will be our batch)
- For each image pair we can choose find matches `[m_a, m_b]` and non-matches `[nm_a, nm_b]`. 
- Run all 2*N images thru the network simultaneously, this is where we can do batch-normalization.
- Compute the loss function as usual and do back-prop.

Things to be careful about.
- Masked matches vs. non-masked matches
